File: chapter04/src/main/java/com/ixiongyu/interfaces/javaversion/Focusable.java
package com.ixiongyu.interfaces.javaversion;

/**
 * @author xiongyu
 * @version Create at ：2020/8/23 下午8:44
 */
public interface Focusable {
    default void showOff() {
        System.out.println("Focusable showOff");
    }

    void setFocus(Boolean focus);
}

File: chapter04/src/main/java/com/ixiongyu/interfaces/Focusable.kt
package com.ixiongyu.interfaces

/**
 * @author  xiongyu
 * @version Create at ：2020/8/23 下午8:40
 */
interface Focusable {
    fun setFocus(focus: Boolean) {
        println("focus:${focus}");
    }

    fun showOff() = println("Focusable showOff")
}
